pkg/intervalpolicy/exponential_backoff_interval_policy.go

```diff
@@ -0,0 +1,38 @@
+package intervalpolicy
+
+import (
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+)
+
+// exponentialBackoffIntervalPolicy is a default interval policy.
+type exponentialBackoffIntervalPolicy struct {
+	exponentialBackoff *backoff.ExponentialBackOff
+	interval           time.Duration
+}
+
+// NewExponentialBackoffIntervalPolicy creates new exponential backoff interval policy.
+func NewExponentialBackoffIntervalPolicy(interval time.Duration) IntervalPolicy {
+	intervalPolicy := &exponentialBackoffIntervalPolicy{exponentialBackoff: backoff.NewExponentialBackOff()}
```

Comment:
this should be created differently.
when we call the function `backoff.NewExponential..`, we use their default values - e.g. the default base factor is 1.5, max value is 60 seconds, randomization factor is 0.5, etc..
do we want every cycle to be multiplied by 1.5? do we want a randomization factor every backoff?
not sure their default values fits for our use case.
anyway, it would be better to put the initialization values ourselves, so we know what to expect.  
we need to create the object with the initialization values we decide/want.
for example:
```
func NewExponentialBackoffIntervalPolicy(interval time.Duration) IntervalPolicy {
	exponentialBackoff := &backoff.ExponentialBackOff{
		InitialInterval:     interval,
		RandomizationFactor: 0, // no use of randomization factor, just multiply by base factor.
		Multiplier:          2, // put as const and call it baseFactor
		MaxInterval:         60 * time.Second, // what is max interval we want? we should probably use some default value and put it as const.. 60s sounds ok to me.
		MaxElapsedTime:      0, // we don't use clock to change the interval, not relevant
		Stop:                0, // we don't use clock to stop the interval, not relevant
		Clock:               backoff.SystemClock,
	}
	exponentialBackoff.Reset()

	return &exponentialBackoffIntervalPolicy{
		exponentialBackoff: exponentialBackoff,
		interval:           interval,
	}
}
```

Reply:
done